app/Helpers/AccountTypeFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿using backend.Models.Accounts;
+
+namespace backend.Helpers
+{
+    public static class AccountTypeFactory
+    {
+        public static AccountTypeEnum CreateAccountTypeEnum(Type type) =>
+            type.Name switch
+            {
+                nameof(PatientModel) => AccountTypeEnum.Patient,
+                nameof(DoctorModel) => AccountTypeEnum.Doctor,
+                nameof(AdminModel) => AccountTypeEnum.Admin,
+                _ => throw new ArgumentException("Invalid account type!")
+            };
+    }
+}
```

Comment:
łamiemy tutaj jakiekolwiek wzorce. Czy nie możemy dodać do AccountModel abstrakcyjnej metody GetEnum() i ją odpowiednio overridować? (tak, możemy https://social.msdn.microsoft.com/Forums/en-US/12e0681d-68ee-4eda-a88c-ff67cbb4d19d/aspnet-mvc-should-we-have-methods-in-view-model-classes?forum=asparchitecture, model może mieć zdefiniowaną metodę publiczną). Wtedy możemy pozbyć się tej fabryki i tylko zrobić user.GetEnum().ToString() w AuthService

Reply:
+1

Reply:
+1

Reply:
Poprawione 👍